test/src/com/xilinx/rapidwright/design/TestNetTools.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2024, Advanced Micro Devices, Inc.
+ * All rights reserved.
+ *
+ * Author: Wenhao Lin, AMD Research and Advanced Development.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.xilinx.rapidwright.design;
+
+import java.util.HashSet;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import com.xilinx.rapidwright.support.RapidWrightDCP;
+
+public class TestNetTools {
+    /**
+     * Tests the method NetTools.isGlobalClock(Net net).
+     * For each DCP file, Vivado uses the TCL command 
+     *      get_nets -hier -parent_net -filter { TYPE == "GLOBAL_CLOCK" } 
+     * to retrieve all nets of type GLOBAL_CLOCK. 
+     * The method NetTools.isGlobalClock(Net net) returns true for these nets and false for others.
+     * 
+     * All Versal/Ultrascale+/Series-7 designs under the RapidWrightDCP path will be checked, except for the following:
+     *      picoblaze4_ooc_X6Y60_X6Y65_X10Y60_X10Y65.dcp    The source cell bufgce_inst of this design is not placed yet.
+     *      picoblaze_ooc_X10Y235_unreadable_edif.dcp       This design would call vivado to generate an readable EDIF file.
+     *      two_clk_check_NetTools.dcp                      This design is not added to the RapidWrightDCP yet.
+     * 
+     * @param pathAndlobalClockNames Use ' ' as the delimiter to split this string, where the first element is the path to the DCP file, 
+     *                               and the remaining elements are the names of the global clock nets reported by Vivado.
+     */
+    @ParameterizedTest
+    @CsvSource({
+            // UltraScale+
```

Comment:
Is there value in testing all of these designs?  I would like to keep the CI load lighter if possible.  Perhaps just 2 or 3 designs?  Alternatively, you could write test code to construct nets on-the-fly and avoid the DCP load costs (see https://github.com/Xilinx/RapidWright/blob/5221b5f1765c7fc8f348b73526b54b4dd4736faa/test/src/com/xilinx/rapidwright/design/TestNet.java#L95 for an example).

Reply:
> Is there value in testing all of these designs?

Yes. 

> I would like to keep the CI load lighter if possible. Perhaps just 2 or 3 designs? Alternatively, you could write test code to construct nets on-the-fly and avoid the DCP load costs (see ...

I would like to take one designs for each series to make the CI load lighter at the final commit. Before that, I hope to ensure that `NetTools` works correctly on all designs in RapidWrightDCP.

Reply:
That seems reasonable.